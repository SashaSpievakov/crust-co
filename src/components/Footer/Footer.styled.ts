import styled, { css } from 'styled-components';

const mixinFooterItems = css`
  display: flex;
  width: fit-content;
  flex-direction: column;
  row-gap: 4px;
`;

const FooterWrapper = styled.footer`
  display: flex;
  justify-content: space-around;
  align-items: end;
  padding: 60px 0;
  row-gap: 50px;

  @media screen and (max-width: 720px) {
    width: fit-content;
    margin: auto;
    flex-direction: column-reverse;
    align-items: stretch;
  }
`;

const Contacts = styled.section`
  ${mixinFooterItems}
`;

const Contact = styled.a`
  width: fit-content;
  border-radius: 5px;

  &:focus-visible {
    background-color: ${({ theme }) => theme.colors.gray};
    outline: 5px solid ${({ theme }) => theme.colors.gray};
    outline-offset: -1px;
  }

  &:hover {
    color: ${({ theme }) => theme.colors.secondary};
  }
`;

const Copyright = styled.section`
  ${mixinFooterItems}
`;

export { Contact, Contacts, Copyright, FooterWrapper };
