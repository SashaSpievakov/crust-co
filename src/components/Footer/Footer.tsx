import { MdEmail, MdPhone, MdLocationOn } from 'react-icons/md';

import { Main, Ul, Link, Icon, Copyright } from './Footer.styled';

const Footer = () => {
  return (
    <Main>
      <Ul>
        <li>
          <Link href="mailto:spievakov@gamil.com">
            <Icon as={MdEmail} />
            <span>spievakov@gmail.com</span>
          </Link>
        </li>

        <li>
          <Link href="tel:+16474739502">
            <Icon as={MdPhone} />
            <span>+1 (647) 473-9502 </span>
          </Link>
        </li>

        <li>
          <Link
            href="https://www.google.com/maps/place/88+Alice+St,+Guelph,+ON+N1E+2Z8/@43.5471712,-80.2392008,17z/data=!3m1!4b1!4m5!3m4!1s0x882b9ae0d0802679:0x74ca5ec44983415f!8m2!3d43.5471673!4d-80.2370121"
            target="_blank"
          >
            <Icon as={MdLocationOn} />
            <span>88 Alice st</span>
          </Link>
        </li>
      </Ul>

      <Copyright>&#169; 2022 Oleksandr Spievakov</Copyright>
    </Main>
  );
};
export default Footer;
