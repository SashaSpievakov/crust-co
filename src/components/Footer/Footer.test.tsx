import '@testing-library/jest-dom';

import rendererWithProviders from '../../tests/helpers/rendererWithProviders';
import Footer from './Footer';

describe('Footer Tests', () => {
  test('renders the Footer component', () => {
    const snapshot = rendererWithProviders(<Footer />);
    expect(snapshot).toMatchSnapshot();
  });
});
