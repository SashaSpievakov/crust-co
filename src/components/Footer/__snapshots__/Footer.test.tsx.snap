// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Tests renders the Footer component 1`] = `
Object {
  "children": Array [
    <ul
      className="sc-gswNZR kZdONu"
    >
      <li>
        <a
          className="sc-dkrFOg huiHRN"
          href="mailto:spievakov@gamil.com"
        >
          <svg
            className="sc-hLBbgP wGpjw"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <span>
            spievakov@gmail.com
          </span>
        </a>
      </li>
      <li>
        <a
          className="sc-dkrFOg huiHRN"
          href="tel:+16474739502"
        >
          <svg
            className="sc-hLBbgP wGpjw"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
            />
          </svg>
          <span>
            +1 (647) 473-9502 
          </span>
        </a>
      </li>
      <li>
        <a
          className="sc-dkrFOg huiHRN"
          href="https://www.google.com/maps/place/88+Alice+St,+Guelph,+ON+N1E+2Z8/@43.5471712,-80.2392008,17z/data=!3m1!4b1!4m5!3m4!1s0x882b9ae0d0802679:0x74ca5ec44983415f!8m2!3d43.5471673!4d-80.2370121"
          target="_blank"
        >
          <svg
            className="sc-hLBbgP wGpjw"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
            />
          </svg>
          <span>
            88 Alice st
          </span>
        </a>
      </li>
    </ul>,
    <p
      className="sc-eDvSVe bWauGy"
    >
      Â© 2022 Oleksandr Spievakov
    </p>,
  ],
  "props": Object {
    "className": "sc-bcXHqe cIQQgr",
  },
  "type": "footer",
}
`;
